web_external/Datasets/UploadDatasetMetadataView.js

```diff
@@ -0,0 +1,248 @@
+// View for a collection of datasets in a select tag
+isic.views.UploadDatasetMetadataSelectDatasetView = isic.View.extend({
```

Comment:
Sorry to do this, but I actually think we should ditch the entire dataset selector box in favor of making this view specific to a single dataset. We could update the routing to accept a dataset ID parameter (which would then be passed to the `UploadDatasetMetadataView`.

I think this has several advantages:
* The `UploadDatasetView` can redirect people directly to a view which adds metadata only to the dataset they just uploaded, which reduces confusion and decision points for newer contributers.
* Since we're only requiring `READ` permissions to upload to a dataset, we should attempt to prevent users from accidentally contributing metadata directly to a public datasetl
* In the future, we may direct users to this dialog via buttons dataset listing view, so we'll be able to go directly to a known dataset ID.